### DISCUSSION TITLE
حل مشکل داکر روی سرور ایران

### DISCUSSION THREAD

**Discussion by Kiya6955** [Gozargah/Marzban#987] [Show and tell]
‌
درود

### برای نصب داکر و حل مشکلات آن روی سرور ایران 

- دستور زیر رو میزنیم تا DNS ها شکن روی سرورمون ست بشه و اعمال بشه.
```
sudo bash -c 'echo -e "nameserver 178.22.122.100\nnameserver 185.51.200.2" > /etc/resolv.conf'
```
- فایل رو ذخیره میکنیم و میایم بیرون سرور رو آپدیت میکنیم و curl و socat رو نصب میکنیم.
```
apt-get update; apt-get upgrade -y; apt-get install curl socat git -y
```
- دستور زیر رو میزنیم تا داکر نصب بشه.
```
curl -fsSL https://get.docker.com | sh
```
- پس از نصب شدن یکی از میرور های داکر رو داخل فایل daemon.json ذخیره میکنیم.
```
nano /etc/docker/daemon.json
```
**IranServer**
```
{
"registry-mirrors": ["https://docker.iranserver.com"]
}
```
**HaioCloud**
```
{
"registry-mirrors": ["https://docker.haiocloud.com"]
}
```
**ParsPack**
```
{
"registry-mirrors": ["https://registry.docker.ir"]
}
```
**AbrArvan**
```
{
  "registry-mirrors": ["https://docker.arvancloud.ir"]
}
```
**Dockeriran**
```
{
   "registry-mirrors": ["https://docker.host:5000"]
}
```
- سپس سرویس داکر را ریستارت کنید.
```
systemctl daemon-reload
systemctl restart docker
```

مشکل داکر رو سرور ایران به طور کامل رفع شد الان میتونید هر چیزی رو میخواین با داکر ران کنید.

#Docker #IRServer #تحریم_داکر

**Comment by iamtheted** [comment]
Merci Kia Jan 💙

**Comment by Habib1385** [comment]
> * پس از نصب شدن یکی از میرور های داکر رو داخل فایل daemon.json ذخیره میکنیم.

سلام 
کیا عزیز مرسی از آموزش خوبت 
این قسمت 
دستور 
`nano /etc/docker/daemon.json`
رو جا انداختی

**Reply by Kiya6955** [reply]
اضافه شد مرسی

**Comment by sarboy1371** [comment]
با این روش مشکلم حل شد و تونستم داکر رو روی سرور ایران نصب کنم
مرسی از کیای عزیز.

**Comment by ghost** [comment]
میتونید از این روش هم استفاده کنید:
در توزیع‌های Debian/Ubuntu فایل زیر را ویرایش نمایید:

```sh
nano /etc/apt/apt.conf
```

سپس خط زیر را به آن اضافه نمایید:

```
Acquire::http::Proxy "http://Proxy.Docker.ir:5555";
```

پس از انجام تنظیم فوق، می توانید مجددا مراحل نصب را ادامه دهید.

**Reply by Kiya6955** [reply]
تست کرده بودم جواب نمیداد برا همین نزاشتم

**Comment by LitenBuzzTh** [comment]
خیبلی عالی بود. بطور کلی برای اتصال به سرورهای آپدیت اوبونتو چطور؟ آیا mirror برای اونم هست بشه یه کاریش کرد؟

**Reply by Kiya6955** [reply]
بله ولی پیشنهاد نمیشه

**Comment by kooshamostofi** [comment]
تشکر فراوان ... من روی ابرآروان این ارور رو میگیرم:
ubuntu@ubuntu-eco-small1-simin-1:~$ apt-get update
Reading package lists... Done
E: Could not open lock file /var/lib/apt/lists/lock - open (13: Permission denied)
E: Unable to lock directory /var/lib/apt/lists/
W: Problem unlinking the file /var/cache/apt/pkgcache.bin - RemoveCaches (13: Permission denied)
W: Problem unlinking the file /var/cache/apt/srcpkgcache.bin - RemoveCaches (13: Permission denied)

سرور نو هستش ممکنه از سرور باشه ؟

**Reply by Kiya6955** [reply]
sudo -i
بزن اول

**Comment by Habib1385** [comment]
> "insecure-registries" : ["https://docker.arvancloud.ir"],

سلام 
کیای عزیز 
توی سرور اروان اگر این خط باشه به مشکل می خوره و داکر ران نمیشه (روی اروان تست کردم)

**Reply by Kiya6955** [reply]
درود ادیت شد

**Comment by hamednaghshinehfard** [comment]
درود
دستور یا اسکریپتی واسه پاک کردن یوزرهای منقضی هم هست واسه مرزبان ؟
با تشکر

**Reply by Kiya6955** [reply]
درود از بات تلگرام خود مرزبان استفاده کنید

**Comment by dev-ir** [comment]
درود نصب داکر بر روی سرورهای ایران به صورت اتوماتیک منتشر شد .
https://github.com/dev-ir/ez-docker

**Comment by baghiyat** [comment]
سلام و عرض ادب

> [root@localhost ~]# apt-get update; apt-get upgrade -y; apt-get install curl socat git -y
Reading package lists... Done
Reading package lists... Done
Building dependency tree... Done
Calculating upgrade... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Reading package lists... Done
Building dependency tree... Done
E: Unable to locate package curl
E: Unable to locate package socat
E: Unable to locate package git

این مشکل را چطوری رفع کنیم؟

**Comment by HosseinTL** [comment]
سلام
میرور جدید ایران سرور با سرعت بالا اضافه شده من باهاش تست کردم خیلی سرعتش بهتره.
{
  "registry-mirrors": ["https://docker.iranserver.com"]
}
اگه میشه اضافه کن همه استفاده کنن.
دمتم گرم

**Comment by AmirMohammadDehghan** [comment]
سلام واسه نصب dokploy روی سرور ایران چیکار باید بکنیم؟

**Comment by fazkal** [comment]
دمت گرم داداش

**Comment by Mahmoodlotfi-1998** [comment]
سلام وقت بخیر
زمان نصب داکپلوی زمان دانلود خود داکپلوی خطای زیر رو میخوره
در حالی که پروکسی پارس پک استفاده کردم.

pull access denied for dokploy/dokploy, repository does not exist or may require 'docker login': denied: <html><body><h1>403 Forbidden</h1>
Since Docker is a US company, we must comply with US export control regulations. In an effort to comply with these, we now block all IP addresses that are located in Cuba, Iran, North Korea, Regions of Ukraine (Crimea, Donetsk, and Luhansk), Sudan, Syria, and Russia. If you are not in one of these cities, countries, or regions and are blocked, please reach out to https://hub.docker.com/support/contact/
</body></html>

**Reply by Kiya6955** [reply]
> سلام وقت بخیر زمان نصب داکپلوی زمان دانلود خود داکپلوی خطای زیر رو میخوره در حالی که پروکسی پارس پک استفاده کردم.
> 
> pull access denied for dokploy/dokploy, repository does not exist or may require 'docker login': denied:
> 
> # 403 Forbidden
> 
> Since Docker is a US company, we must comply with US export control regulations. In an effort to comply with these, we now block all IP addresses that are located in Cuba, Iran, North Korea, Regions of Ukraine (Crimea, Donetsk, and Luhansk), Sudan, Syria, and Russia. If you are not in one of these cities, countries, or regions and are blocked, please reach out to https://hub.docker.com/support/contact/

درود مطمعن شو داکر ریستارت کردی.
docker login 
ام بزن لاگین کن یکبار

**Reply by ghissari** [reply]
راهکار گفته شده فقط برای دانلود کردن ایمیج های پابلیک جواب میده و برای لاگین کردن در داکرهاب یا پول کردن ایمیج های پرایوت کاربردی نداره

**Reply by Kiya6955** [reply]
> راهکار گفته شده فقط برای دانلود کردن ایمیج های پابلیک جواب میده و برای لاگین کردن در داکرهاب یا پول کردن ایمیج های پرایوت کاربردی نداره

DNS برای همونه خب!

**Comment by soshyant-joshaghani** [comment]
sudo sed -i 's/archive.ubuntu.com/ir.archive.ubuntu.com/g' /etc/apt/sources.list.d/ubuntu.sources
sudo sed -i 's/security.ubuntu.com/ir.archive.ubuntu.com/g' /etc/apt/sources.list.d/ubuntu.sources

این دستور هم میرور های ایران رو برای اوبونتو اضافه میکنه به ذهنم رسید بگم

**Comment by MohammadHeydari** [comment]
Well done @Kiya6955
